calico_test/tests/st/utils/route_reflector.py

```diff
@@ -47,21 +48,41 @@ def __enter__(self):
                 rr = DockerHost('RR.%d.%d' % (ii, jj), start_calico=False)
                 ip = "-e IP=%s" % rr.ip
                 rr.execute("docker load --input /code/routereflector.tar")
-                rr.execute("docker run --privileged --net=host -d "
-                           "--name rr %s %s "
-                           "calico/routereflector" % (etcd_auth, ip))
 
-                # Invoke the suggested curl command to add the RR entry to
-                # etcd.
+                # Check which type of etcd is being run, then invoke the
+                # suggested curl command to add the RR entry to etcd.
                 #
                 # See https://github.com/projectcalico/calico-bird/tree/feature-ipinip/build_routereflector
                 # for details.
-                rr.execute(r'curl -L http://%s:2379/v2/keys/calico/bgp/v1/rr_v4/%s '
-                           r'-XPUT -d value="{'
-                             r'\"ip\":\"%s\",'
-                             r'\"cluster_id\":\"%s\"'
-                           r'}"' % (get_ip(), rr.ip, rr.ip, cluster_id))
+                if os.getenv("ETCD_SCHEME", None) == "https":
+                    # Etcd is running with SSL/TLS, pass the key values
+                    etcd_ca = "-e ETCD_CA_CERT_FILE=%s" % ETCD_CA
+                    etcd_cert = "-e ETCD_CERT_FILE=%s" % ETCD_CERT
+                    etcd_key = "-e ETCD_KEY_FILE=%s" % ETCD_KEY
+                    rr.execute("docker run --privileged --net=host -d "
```

Comment:
Side note:  You also need to update the README.md doc in calico-bird that describes the RR.